tests/runtime.rs

```diff
@@ -72,78 +77,130 @@ fn runtime_single_threaded_block_on() {
     tokio::runtime::current_thread::block_on_all(create_client_server_future()).unwrap();
 }
 
-#[test]
-fn runtime_single_threaded_block_on_all() {
-    let cnt = Arc::new(Mutex::new(0));
-    let c = cnt.clone();
+mod runtime_single_threaded_block_on_all {
+    use super::*;
 
-    let msg = tokio::runtime::current_thread::block_on_all(lazy(move || {
-        {
-            let mut x = c.lock().unwrap();
-            *x = 1 + *x;
-        }
+    fn test<F>(spawn: F)
+    where
+        F: Fn(Box<Future<Item=(), Error=()> + Send>),
+    {
+        let cnt = Arc::new(Mutex::new(0));
+        let c = cnt.clone();
 
-        // Spawn!
-        tokio::spawn(lazy(move || {
+        let msg = tokio::runtime::current_thread::block_on_all(lazy(move || {
             {
                 let mut x = c.lock().unwrap();
                 *x = 1 + *x;
             }
-            Ok::<(), ()>(())
-        }));
 
-        Ok::<_, ()>("hello")
-    })).unwrap();
+            // Spawn!
+            spawn(Box::new(lazy(move || {
+                {
+                    let mut x = c.lock().unwrap();
+                    *x = 1 + *x;
+                }
+                Ok::<(), ()>(())
+            })));
 
-    assert_eq!(2, *cnt.lock().unwrap());
-    assert_eq!(msg, "hello");
+            Ok::<_, ()>("hello")
+        })).unwrap();
+
+        assert_eq!(2, *cnt.lock().unwrap());
+        assert_eq!(msg, "hello");
+    }
+
+    #[test]
+    fn spawn() {
+       test(|f| { tokio::spawn(f); })
+    }
+
+    #[test]
+    fn execute() {
+        test(|f| {
+            tokio::executor::DefaultExecutor::current()
+                .execute(f)
+                .unwrap();
+        })
+    }
 }
 
-#[test]
-fn runtime_single_threaded_racy_spawn() {
-    let (trigger, exit) = futures::sync::oneshot::channel();
-    let (handle_tx, handle_rx) = ::std::sync::mpsc::channel();
-    let jh = ::std::thread::spawn(move || {
-        let mut rt = tokio::runtime::current_thread::Runtime::new().unwrap();
-        handle_tx.send(rt.handle()).unwrap();
-
-        // don't exit until we are told to
-        rt.block_on(exit.map_err(|_| ())).unwrap();
-
-        // run until all spawned futures (incl. the "exit" signal future) have completed.
-        rt.run().unwrap();
-    });
+mod runtime_single_threaded_racy {
+    use super::*;
+    fn test<F>(spawn: F)
+    where
+        F: Fn(
+            tokio::runtime::current_thread::Handle,
+            Box<Future<Item=(), Error=()> + Send>,
+        ),
+    {
+        let (trigger, exit) = futures::sync::oneshot::channel();
+        let (handle_tx, handle_rx) = ::std::sync::mpsc::channel();
+        let jh = ::std::thread::spawn(move || {
+            let mut rt = tokio::runtime::current_thread::Runtime::new().unwrap();
+            handle_tx.send(rt.handle()).unwrap();
+
+            // don't exit until we are told to
+            rt.block_on(exit.map_err(|_| ())).unwrap();
```

Comment:
IFAIK rt.block_on does not need to `.map_err(|_|())`

Reply:
It does not, but it is a minor issue. You can update in a PR if you wish.